plugins/browser-plugin-vimeo-tracking/src/mediaPluginBinding.ts

```diff
@@ -0,0 +1,215 @@
+      trackedPlayer.player
+        .getVolume()
+        .then((newVolume) => trackMediaVolumeChange({ id, newVolume: Math.round(newVolume * 100) }))
+        .catch(trackError);
+      break;
+
+    case InternalVimeoEvent.FULLSCREENCHANGE:
+      trackedPlayer.player
+        .getFullscreen()
+        .then((fullscreen) => trackMediaFullscreenChange({ id, fullscreen }))
+        .catch(trackError);
+      break;
+
+    case InternalVimeoEvent.PLAYBACKRATECHANGE:
+      trackedPlayer.player
+        .getPlaybackRate()
+        .then((newRate) => trackMediaPlaybackRateChange({ id, newRate }))
+        .catch(trackError);
+      break;
+
+    case InternalVimeoEvent.ENTERPICTUREINPICTURE:
+      trackMediaPictureInPictureChange({ id, pictureInPicture: true });
+      break;
+
+    case InternalVimeoEvent.LEAVEPICTUREINPICTURE:
+      trackMediaPictureInPictureChange({ id, pictureInPicture: false });
+      break;
+
+    case InternalVimeoEvent.QUALITYCHANGE:
+      trackMediaQualityChange({ id, newQuality: (eventData as QualityChangeEvent).quality });
+      break;
+
+    case InternalVimeoEvent.CUEPOINT:
+      // `time` can be a keyword in some warehouses, so we rename to `cuePointTime`
+      (eventData as VimeoPlayerCuePointData).cuePointTime = (eventData as VimeoCuePointData).time;
+      delete (eventData as VimeoCuePointData).time;
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.CUEPOINT,
+          data: eventData as VimeoPlayerCuePointData,
+        },
+      });
+      break;
+
+    case InternalVimeoEvent.TEXTTRACKCHANGE:
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.TEXT_TRACK_CHANGE,
+          data: eventData as TextTrackData,
+        },
+      });
+      break;
+
+    case InternalVimeoEvent.INTERACTIVEHOTSPOTCLICKED:
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.HOTSPOT_CLICK,
+          data: {},
+        },
+        context: [{ schema: VimeoSchema.INTERACTION, data: eventData as VimeoInteractiveHotspotClickedData }],
+      });
+      break;
+
+    case InternalVimeoEvent.INTERACTIVEOVERLAYPANELCLICKED:
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.OVERLAY_CLICK,
+          data: {},
+        },
+        context: [{ schema: VimeoSchema.INTERACTION, data: eventData as VimeoInteractiveHotspotClickedData }],
+      });
+      break;
+
+    case InternalVimeoEvent.CHAPTERCHANGE:
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.CHAPTER_CHANGE,
+          data: eventData as ChapterData,
+        },
+      });
+      break;
+
+    case InternalVimeoEvent.LOADED:
+      // 'Loaded' can be fired by a new video being loaded into the same player
+      // so we should ensure player attributes are up-to-date
+      getVimeoMediaAttributes(trackedPlayer)
```

Comment:
Nit: Is this necessary given that we already did the same thing in the `trackEvent` call?